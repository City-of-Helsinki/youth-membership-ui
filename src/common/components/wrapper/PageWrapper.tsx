import React, { PropsWithChildren } from 'react';
import { Helmet } from 'react-helmet-async';
import { useTranslation } from 'react-i18next';

type Props = PropsWithChildren<{}>;

function PageWrapper(props: Props) {
  const { t } = useTranslation();
  return (
    <>
      <Helmet>
        <title>{t('appName')}</title>
        <meta name="description" content={t('meta.description')} />
        <meta property="og:title" content={t('appName')} />
        <meta property="og:description" content={t('meta.description')} />
      </Helmet>
      {props.children}
    </>
  );
}

export default PageWrapper;
