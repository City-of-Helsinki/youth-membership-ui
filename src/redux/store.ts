import {
  configureStore,
  getDefaultMiddleware,
  Action,
  ConfigureStoreOptions,
} from 'redux-starter-kit';
import { ThunkAction } from 'redux-thunk';

import rootReducer, { RootState } from './rootReducer';

const isProd = process.env.NODE_ENV === 'production';

export type AppThunk = ThunkAction<void, RootState, null, Action<string>>;

export function configureAppStore(options: Partial<ConfigureStoreOptions>) {
  const store = configureStore({
    reducer: rootReducer,
    ...options,
  });

  /* eslint-disable @typescript-eslint/no-explicit-any */
  if (!isProd && (module as any).hot) {
    (module as any).hot.accept('./rootReducer', () => {
      store.replaceReducer(rootReducer);
    });
  }
  /* eslint-enable @typescript-eslint/no-explicit-any */

  return store;
}

const store = configureAppStore({
  devTools: !isProd,
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export default store;
