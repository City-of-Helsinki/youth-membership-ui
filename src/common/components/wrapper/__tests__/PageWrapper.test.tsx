import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HelmetProvider } from 'react-helmet-async';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';

import store from '../../../../redux/store';
import PageWrapper from '../PageWrapper';

it('matches snapshot', () => {
  const wrapper = mount(
    <Provider store={store}>
      <HelmetProvider>
        <MemoryRouter keyLength={0}>
          <PageWrapper />
        </MemoryRouter>
      </HelmetProvider>
    </Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
