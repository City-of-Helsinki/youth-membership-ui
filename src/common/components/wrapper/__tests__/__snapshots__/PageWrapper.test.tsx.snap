// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Jässäri
    </title>
    <meta
      content="Jässärillä pääset mukaan nuorisopalveluiden toimintaan. Jäsenyys takaa sinulle kivaa tekemistä, voit hengailla nuorisotaloilla, osallistua harrastusryhmiin, kursseille ja leireille, hakea projektiavustusta tai vaikka tuottaa oman tapahtuman! Nuorisotaloilla toimintaa ohjaavat koulutetut nuoriso-ohjaajat."
      name="description"
    />
    <meta
      content="Jässäri"
      property="og:title"
    />
    <meta
      content="Jässärillä pääset mukaan nuorisopalveluiden toimintaan. Jäsenyys takaa sinulle kivaa tekemistä, voit hengailla nuorisotaloilla, osallistua harrastusryhmiin, kursseille ja leireille, hakea projektiavustusta tai vaikka tuottaa oman tapahtuman! Nuorisotaloilla toimintaa ohjaavat koulutetut nuoriso-ohjaajat."
      property="og:description"
    />
  </Helmet>
</Fragment>
`;
